modules/javafx.controls/src/test/java/test/com/sun/javafx/scene/control/LambdaMultipleHandlerTest.java

```diff
@@ -0,0 +1,255 @@
+        handler.registerListChangeListener(items, consumer);
+        handler.registerListChangeListener(otherList, consumer);
+        items.add("added");
+        otherList.add("added other");
+        assertEquals(2, changes.size());
+        assertEquals(items, changes.get(0).getList());
+        assertEquals(otherList, changes.get(1).getList());
+    }
+
+    /**
+     * Test that multiple consumers to same observable are invoked in order
+     * of registration.
+     */
+    @Test
+    public void testRegisterMultipleConsumerToSingleList() {
+        List<Change<?>> changes = new ArrayList<>();
+        Consumer<Change<?>> consumer = change -> changes.add(change);
+        List<Change<?>> secondChanges = new ArrayList<>();
+        Consumer<Change<?>> secondConsumer = change -> secondChanges.addAll(changes);
+        handler.registerListChangeListener(items, consumer);
+        handler.registerListChangeListener(items, secondConsumer);
+        items.add("added");
+        assertEquals(1, changes.size());
+        assertEquals(changes, secondChanges);
+    }
+
+    @Test
+    public void testRegister() {
+        List<Change<?>> changes = new ArrayList<>();
+        Consumer<Change<?>> consumer = change -> changes.add(change);
+        handler.registerListChangeListener(items, consumer);
+        String added = "added";
+        items.add(added);
+        assertEquals(1, changes.size());
+        Change<?> change = changes.get(0);
+        change.next();
+        assertTrue(change.wasAdded());
+        assertTrue(change.getAddedSubList().contains(added));
+    }
+
+    @Test
+    public void testRegisterNullConsumer() {
+        handler.registerListChangeListener(items, null);
+    }
+
+    @Test
+    public void testRegisterNullList() {
+        handler.registerListChangeListener(null, c -> {});
+    }
+
+//--------- dispose
+
+    @Test
+    public void testDispose() {
+        List<Change<?>> changes = new ArrayList<>();
+        Consumer<Change<?>> consumer = change -> changes.add(change);
+        handler.registerListChangeListener(items, consumer);
+        handler.dispose();
+        items.add("added");
+        assertEquals("listener must not be invoked after dispose", 0, changes.size());
+        handler.registerListChangeListener(items, consumer);
+        items.add("added");
+        assertEquals("listener must be invoked when re-registered after dispose", 1, changes.size());
+    }
+
+
+//--------- test weak registration
+
+    /**
+     * Test that handler is gc'ed and listener no longer notified.
+     */
+    @Test
+    public void testRegisterMemoryLeak() {
+        List<Change<?>> changes = new ArrayList<>();
+        Consumer<Change<?>> consumer = change -> changes.add(change);
+        WeakReference<LambdaMultiplePropertyChangeListenerHandler> ref =
+                new WeakReference<>(new LambdaMultiplePropertyChangeListenerHandler());
+        ref.get().registerListChangeListener(items, consumer);
```

Comment:
changed as suggested. I used a similar pattern in BehaviorMemoryTest - should I file a testbug to change it in the same way?

Reply:
Yes, please file a new bug (not high priority, since it is a corner case). I took a look and there are a few other controls tests that use this pattern as well (BehaviorCleanupTest, ListViewTest, TabPaneTest, SkinCleanupTest, SkinLabeledCleanupTest, SkinMemoryLeakTest), so the new bug could cover fixing this for all of them.

Reply:
done and taken (should cleanup my own dirt ;) [JDK-8265406](https://bugs.openjdk.java.net/browse/JDK-8265406)